### PR TITLE
Calls the function Calendar when a incident has resolved

### PR DESCRIPTION
**Pull request by NFavareto**
### Description
Calls the function Calendar inside  calendar's package ,when a incident has resolved by `/hellper_resolve`

### How to test?
Resolved a incident using the command `/hellper_resolve`.

image

### Expected behavior
Everything should work as expected and the Post Mortem meeting should have been scheduled on your Google Calendar.

### CODE REVIEW

File: internal/commands/resolve.go [hunk 1]
```diff
@@ -40,6 +42,28 @@ func ResolveIncidentDialog(client bot.Client, triggerID string) error {
 		Subtype: slack.InputSubtypeURL,
 	}
 
+	postMortemMeeting := &slack.DialogInputSelect{
+		DialogInput: slack.DialogInput{
+			Label:       "Can i schedule a meeting of Post Mortem?",
```
Comment (GLOFonseca) [inline internal/commands/resolve.go:47]:
```suggestion
			Label:       "Can I schedule a Post Mortem meeting?",
```

File: internal/commands/resolve.go [hunk 2]
```diff
@@ -40,6 +42,28 @@ func ResolveIncidentDialog(client bot.Client, triggerID string) error {
...
+			Name:        "post_mortem_meeting",
+			Type:        "select",
+			Placeholder: "Post Mortem Meeting",
+			Optional:    false,
```
Comment (GLOFonseca) [inline internal/commands/resolve.go:51]: Why we don't set this as an optional field, and the default is false?

File: internal/commands/resolve.go [hunk 3]
```diff
@@ -86,8 +113,19 @@ func ResolveIncidentByDialog(ctx context.Context, client bot.Client, logger log.
...
+	isPostMortemMeeting, err := strconv.ParseBool(postMortemMeeting)
+	if err != nil {
+		return err
+	}
+
+	if isPostMortemMeeting {
+		//postMortemMeetingURL = calendar.CreateCalendarEvent() [locked - card do feijo ]aqui vai ser o tratamento de erros.
+		calendar.CreateCalendarEvent()
+		postMortemMeetingURL = "link ficticio"
+	}
```
Comment (GLOFonseca) [inline internal/commands/resolve.go:125]: In our PR we are not returning a string here, but a struct that contains information about the PM meeting. One of the fields is the URL.

File: internal/commands/resolve.go (hunk 2 above)
Comment (NFavareto) [reply internal/commands/resolve.go:52]: As it is optional, it allows the user not to fill in a value. When the user sends nothing, the default value is not taken into account by the Dialog component.

image

The default value is used when filling in the field. So I left it as mandatory.
image

File: internal/commands/resolve.go [hunk 4]
```diff
@@ -105,14 +159,41 @@ func ResolveIncidentByDialog(ctx context.Context, client bot.Client, logger log.
...
+	switch previewPostMortemDate.Weekday() {
+	case time.Saturday:
+		postMortemGapDays += 2
+	case time.Sunday:
+		postMortemGapDays++
+	default:
+		break
+	}
+
+	utc, _ := time.LoadLocation(timezone)
```
Comment (flucas97) [inline internal/commands/resolve.go:174]: Just remembering, this error ignored will be handled in v2

File: internal/commands/resolve.go (hunk 2 above)
Comment (GLOFonseca) [reply internal/commands/resolve.go:52]: Agreed

File: internal/commands/resolve.go [hunk 5]
```diff
@@ -136,15 +217,26 @@ func createResolveChannelAttachment(inc model.Incident, userName string) slack.A
...
 	}
 }
 
-func createResolvePrivateAttachment(inc model.Incident) slack.Attachment {
+func createResolvePrivateAttachment(inc model.Incident, postMortemMeetingURL string) slack.Attachment {
 	var privateText strings.Builder
 	privateText.WriteString("The Incident <#" + inc.ChannelId + "> has been resolved by you\n\n")
 	privateText.WriteString("*Status.io:* Be sure to update the incident status on" + inc.StatusPageUrl + "\n")
-	privateText.WriteString("*Post Mortem:* Don't forget to bookmark Post Mortem for the incident <#" + inc.ChannelId + ">\n\n")
+	privateText.WriteString("*Post Mortem:* Don't forget to bookmark Post Mortem for the incident <#" + inc.ChannelId + ">\n")
```
Comment (GLOFonseca) [inline internal/commands/resolve.go:232]: This message remembers the commander to bookmark the meeting, so to create a new event.
As this meeting is not obligatory now, you can check if that is a URL, if so inform it, if not, only remember to fill in the postmortem document.

File: internal/commands/resolve.go [hunk 6]
```diff
@@ -136,15 +217,26 @@ func createResolveChannelAttachment(inc model.Incident, userName string) slack.A
...
+	if postMortemMeetingURL != "" {
+		privateText.WriteString("*Post Mortem Meeting Link:* Don't forget you are scheduled the Post Mortem Meeting <`" + postMortemMeetingURL + "`>\n\n")
+		postMortemMeetingURL = "Don't forget you are scheduled the Post Mortem Meeting<`" + postMortemMeetingURL + "`>"
```
Comment (GLOFonseca) [inline internal/commands/resolve.go:235]: I did not understand why to change the variable

File: internal/commands/resolve.go [hunk 7]
```diff
@@ -136,15 +217,26 @@ func createResolveChannelAttachment(inc model.Incident, userName string) slack.A
...
 
-func createResolvePrivateAttachment(inc model.Incident) slack.Attachment {
+func createResolvePrivateAttachment(inc model.Incident, postMortemMeetingURL string) slack.Attachment {
 	var privateText strings.Builder
 	privateText.WriteString("The Incident <#" + inc.ChannelId + "> has been resolved by you\n\n")
 	privateText.WriteString("*Status.io:* Be sure to update the incident status on" + inc.StatusPageUrl + "\n")
-	privateText.WriteString("*Post Mortem:* Don't forget to bookmark Post Mortem for the incident <#" + inc.ChannelId + ">\n\n")
+	privateText.WriteString("*Post Mortem:* Don't forget to bookmark Post Mortem for the incident <#" + inc.ChannelId + ">\n")
+	if postMortemMeetingURL != "" {
+		privateText.WriteString("*Post Mortem Meeting Link:* Don't forget you are scheduled the Post Mortem Meeting <`" + postMortemMeetingURL + "`>\n\n")
```
Comment (GLOFonseca) [inline internal/commands/resolve.go:234]: This phrase is confusing, what about just:

```suggestion
		privateText.WriteString("*Post Mortem Meeting Link:* <`" + postMortemMeetingURL + "`>\n\n")
```

File: internal/commands/resolve.go (hunk 4 above)
Comment (GLOFonseca) [reply internal/commands/resolve.go:174]: Even if the proper handling will be done in the future, you can return in a `err` variable, and at least return it, or even log it.

File: internal/commands/resolve.go [hunk 8]
```diff
@@ -86,8 +117,31 @@ func ResolveIncidentByDialog(ctx context.Context, client bot.Client, logger log.
...
 	}
 
-	channelAttachment := createResolveChannelAttachment(incident, userName)
-	privateAttachment := createResolvePrivateAttachment(incident)
+	isPostMortemMeeting, err := strconv.ParseBool(postMortemMeeting)
+	if err != nil {
+		return err
+	}
+
+	if isPostMortemMeeting {
```
Comment (GLOFonseca) [inline internal/commands/resolve.go:125]: What do you think about remove this condition to a private method?

Comment (NFavareto) [reply internal/commands/resolve.go:134]: I dont understood this propose :thinking:

Comment (GLOFonseca) [reply internal/commands/resolve.go:134]: I think everything inside this if could be inside a private method but is just a provocation, it's not necessary.

File: internal/commands/resolve.go [hunk 9]
```diff
@@ -86,8 +117,31 @@ func ResolveIncidentByDialog(ctx context.Context, client bot.Client, logger log.
 		return err
 	}
 
-	channelAttachment := createResolveChannelAttachment(incident, userName)
-	privateAttachment := createResolvePrivateAttachment(incident)
+	isPostMortemMeeting, err := strconv.ParseBool(postMortemMeeting)
```
[the remark this answers is not in the document]
Comment (GLOFonseca) [reply internal/commands/resolve.go:124]: @NFavareto what did you thought about it?

File: internal/commands/resolve.go [hunk 10]
```diff
@@ -86,8 +117,31 @@ func ResolveIncidentByDialog(ctx context.Context, client bot.Client, logger log.
...
+	if err != nil {
+		return err
+	}
+
+	if isPostMortemMeeting {
+		finishDate := incident.EndTimestamp
+		startMeeting, endMeeting := setMeetingDate(finishDate, postMortemGapDays, timezone)
+		summary := "[Post Mortem] " + channelName
+		emails := []string{}
```
[the remark this answers is not in the document]
Comment (GLOFonseca) [reply internal/commands/resolve.go:147]: @NFavareto I think we can add this comment, as probably the Pause epic will have priority over the due date V2

File: internal/commands/resolve.go [hunk 11]
```diff
@@ -86,8 +117,31 @@ func ResolveIncidentByDialog(ctx context.Context, client bot.Client, logger log.
...
+
+		user, err := getSlackUserInfo(ctx, client, logger, userID)
+		if err != nil {
+			return fmt.Errorf("commands.ResolveIncidentByDialog.get_slack_user_info: incident=%v commanderId=%v error=%v", channelID, userID, err)
+		}
+
+		calendarEvent, err := calendar.CreateCalendarEvent(ctx, startMeeting, endMeeting, summary, user.Email, emails)
```
[the remark this answers is not in the document]
Comment (GLOFonseca) [reply internal/commands/resolve.go:159]: @NFavareto what do you think about it?

File: internal/commands/resolve.go [hunk 12]
```diff
@@ -86,8 +117,31 @@ func ResolveIncidentByDialog(ctx context.Context, client bot.Client, logger log.
...
-	channelAttachment := createResolveChannelAttachment(incident, userName)
-	privateAttachment := createResolvePrivateAttachment(incident)
+	isPostMortemMeeting, err := strconv.ParseBool(postMortemMeeting)
+	if err != nil {
+		return err
+	}
+
+	if isPostMortemMeeting {
+		finishDate := incident.EndTimestamp
+		startMeeting, endMeeting := setMeetingDate(finishDate, postMortemGapDays, timezone)
```
[the remark this answers is not in the document]
Comment (GLOFonseca) [reply internal/commands/resolve.go]: I think it is not necessary to use a new variable that will be used only in the next line

File: development.env.example [hunk 13]
```diff
@@ -21,3 +21,5 @@ HELLPER_VERIFICATION_TOKEN=YOUR_SLACK_VERIFICATION_TOKEN
 HELLPER_NOTIFY_ON_RESOLVE=true
 HELLPER_NOTIFY_ON_CLOSE=true
 FILE_STORAGE=google_drive
+HELLPER_PRODUCT_LIST=Product A;Product B;Product C
+TIMEZONE=America/Sao_Paulo
```
Comment (tiagoemsi) [inline development.env.example:25]: I missed this environment variable in the documentation.
All environment variables must also be in:
* README.md
* app.json

Comment (NFavareto) [reply development.env.example:25]: Done!

**Pull request merged**